### PR TITLE
Remove html stripping bug from html.mako

### PR DESCRIPTION
**Pull request by IAlwaysBeCoding**
Removes a bug where it strips the first character of every line in the source code if you leave first line blank.

### CODE REVIEW

**Comment by IAlwaysBeCoding** [comment]
@BurntSushi  I like your solution better

**Comment by IAlwaysBeCoding** [comment]
Well, am reopening this PR to fix #84 .

File: pdoc/templates/html.mako [hunk 1]
```diff
@@ -38,6 +38,7 @@
     Returns one string with all of the source code.
     """
     base_indent = len(indent.match(lines[0]).group(0))
+    base_indent = 0 if lines[0].isspace() else base_indent
```
Comment (BurntSushi) [inline pdoc/templates/html.mako]: Could you change this to:

```python
base_indent = 0
for line in line:
    if len(line.strip()) > 0:
        base_indent = len(indent.match(lines[0]).group(0))
        break
```

(Assuming it works. :-))

**Comment by IAlwaysBeCoding** [comment]
Okay, got it its fixed. You just had a small spelling mistake,  in the `for line in line`. Other than that it works.

**Comment by BurntSushi** [comment]
Yay! @IAlwaysBeCoding thank you! And thanks for your patience. :-)

Enjoy!

**Comment by IAlwaysBeCoding** [comment]
Oh, no ! Thank you, am kinda new to this thing of doing PR's . Thanks for letting me do a PR. I liked that little interaction we had there :+1: . Also, am curious since you seem to write a lot in the Rust language, how would you do a quick comparison between pdoc(python) and Docopt(rust) . Eventually, am going to read that one next.As for my new years resolution,I am trying start another big journey like I did with python a few years ago(focus on one language as my primary language), and I wanted to get an opinion of someone who came from a python background and picked up Rust extensively.

**Comment by BurntSushi** [comment]
@IAlwaysBeCoding I <3 Rust. (I'm a [member of the library team](https://www.rust-lang.org/team.html#Library).) Docopt(rust) is actually a command line argument parser in the style of [docopt.org](http://docopt.org/) (whose reference implementation is actually written in Python). So the two projects aren't particularly comparable.

With that said, Rust is a great language to learn. If you don't have a systems language background (e.g., C, C++), then it might be a little intimidating at first, but there are plenty of folks learning Rust without systems language experience. Plus, we're all very friendly with beginners. :-)

Generally speaking, the class of problems one might want to solve with Python as opposed to Rust is going to be very different. (Not to say there isn't any overlap though!) Rust is going to give you a static type system, which means you'll need to get your code past the compiler before it can run. Rust also doesn't have a garbage collector, so you'll need to reason about the lifetime of data in your program. It can be a little daunting at first, but the [Rust book](https://doc.rust-lang.org/stable/book/) is a great place to start.

IRC is a wonderful resource. I'm usually around when I'm not at work. :-)

**Comment by IAlwaysBeCoding** [comment]
@BurntSushi awesome! thanks for the reply. I will take a look at the rust IRC.I really like their cargo system they have, much better than python pip if you ask me. I hardly install anything with direct pip from the main repository. I always install from github(on the condition there is a setup.py there) as it has proven much better for stability.

**Pull request merged**